### PR TITLE
Chore: setup postgres and vector in dockerfile

### PR DESCRIPTION
**Pull request by StrongMonkey**
This PR should configure postgres and pgvector into otto dockerfile. It will run and use the built-in postgres process if no OTTO8_SERVER_DSN is set.

I ran some test against knowledge ingestion and it works fine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,35 +22,33 @@ RUN --mount=type=cache,id=pnpm,target=/root/.local/share/pnpm/store \
     UV_LINK_MODE=copy BIN_DIR=/bin make package-tools
 
 FROM cgr.dev/chainguard/wolfi-base AS final
+
+# Setup postgres and postgres-vector, although in production setup we don't really need standalone postgres
+RUN apk add postgresql build-base git postgresql-dev
+RUN git clone --branch v0.8.0 https://github.com/pgvector/pgvector.git && \
```
Comment (cjellick) [inline Dockerfile:26]: darren mentioned this fleetingly when the feasibility of this was discussed. i should have passed it along sooner. he said you could just get it via from, like
```
FROM pgvector/pgvector:pg17
```
or whatever. i think that would be a little cleaner and more official feeling?

Comment (StrongMonkey) [reply Dockerfile:26]: yep, I did detour to avoid dropping wolfi-base image we already uses.

**Review by thedadams — APPROVED** [review]

**Comment by StrongMonkey** [comment]
Did the following change based on feedback
1. Move the pgvector build stage into a separate step. And copy the necessary artifacts to the final image after it is done.
2. Use `cgr.dev/chainguard/postgres:latest-dev` as the final image. Noticed that we can't use `cgr.dev/chainguard/postgres:latest` because it is a `distroless` image that doesn't come with `apk`
3. Use `/data/postgresql` as data store for postgresql.

**Review by iwilltry42 — APPROVED** [review]

**Comment by cjellick** [comment]
Looks good. I'd like a rereview by @thedadams or a review by @ibuildthecloud

**Review by ibuildthecloud — APPROVED** [review]
Assuming it works...

**Comment by StrongMonkey** [comment]
@ibuildthecloud come on where is the trust :)

**Pull request merged**